### ISSUE TITLE
Client not pickingup correct role on EKS node with ServiceAccount applied.

### ISSUE THREAD

**Issue by hardillb** [aws/aws-sdk-js-v3#2368]
## Describe the bug
Appear to be using the wrong role/credentials with a service account

I have a role bound to a Kubernetes Service account on EKS, which when applied to a pod is setting the following environment variables:
- AWS_DEFAULT_REGION:           us-east-2
- AWS_REGION:                   us-east-2
- AWS_ROLE_ARN:                 arn:aws:iam::202058804932:role/eksctl-node-red-addon-iamserviceaccount-defa-Role1-16K0LE4QTM51J
- AWS_WEB_IDENTITY_TOKEN_FILE:  /var/run/secrets/eks.amazonaws.com/serviceaccount/token

Policy:
```
{
    "Version": "2012-10-17",
    "Statement": [
        {
            "Effect": "Allow",
            "Action": [
                "elasticfilesystem:CreateAccessPoint",
                "elasticfilesystem:DeleteAccessPoint"
            ],
            "Resource": "*"
        }
    ]
}
```

It looks like the client is picking up the generic role, not the one being passed in from the service account.

#### SDK version number
`@aws-sdk/client-efs@3.14.0`

#### Is the issue in the browser/Node.js/ReactNative?
Node.js

#### Details of the browser/Node.js/ReactNative version
```
/usr/src/app # node -v
v12.20.1
```

## Steps to reproduce
I followed these instructions to set up a policy/role/ServiceAccount mapping:

https://docs.aws.amazon.com/eks/latest/userguide/iam-roles-for-service-accounts.html

I run the following code:
```
const Storage = require('./storage.js`)
const storage = new Storage("us-east-2", "fs-xxxxxx")
stroage.CreateAccessPoint("foo")
```
Where `storage.js` is:

```
const { EFSClient, 
	DescribeAccessPointsCommand,
	CreateAccessPointCommand,
	DeleteAccessPointCommand } = require('@aws-sdk/client-efs')

function Storage(fsID, region) {
	this._fsID = fsID
	this._efsClient = new EFSClient()
}

Storage.prototype.CreateAccessPoint = function(name){
	const params = {
		FileSystemId: this._fsID,
		PosixUser: {
			Uid: 1000,
			Gid: 1000
		},
		RootDirectory: {
			Path: "/"+ name,
			CreationInfo: {
				OwnerUid: 1000,
				OwnerGid: 1000,
				Permissions: "755"
			}
		},
		Tags: [
			{Key: "Name", Value: name+"-ap"}
		]
	}

	const apCommand = new CreateAccessPointCommand(params)
	return this._efsClient.send(apCommand)
}

Storage.prototype.DeleteAccessPoint = function(name){
	const params = {
		FileSystemId: this._fsID
	}
	const describeCommand = new DescribeAccessPointsCommand(params)
	return this._efsClient.send(describeCommand)
	.then(aps => {
		if (aps) {
			for (var i=0; i<aps.AccessPoints.length; i++) {
				let ap = aps.AccessPoints[i]
				if (ap.Name == name + "-ap" ) {
					const delParam = {
						AccessPointId: ap.AccessPointId
					}
					const delCommand = new DeleteAccessPointCommand(delParam)
					return this._efsClient.send(delCommand)
				}
			}
		}
	})
}


module.exports = Storage
```

I've tried with both as is, and with passing `{region: region}` to the client constructor.

#### Observed behavior
I get the following error
```
AccessDeniedException: User: arn:aws:sts::202058804932:assumed-role/eksctl-node-red-nodegroup-ng-0fe4-NodeInstanceRole-1ROIX18NYR233/i-0ee6c7874944f5784 is not authorized to perform: elasticfilesystem:CreateAccessPoint on the specified resource
    at deserializeAws_restJson1CreateAccessPointCommandError (/usr/src/app/node_modules/@aws-sdk/client-efs/dist/cjs/protocols/Aws_restJson1.js:857:41)
    at processTicksAndRejections (internal/process/task_queues.js:97:5)
    at async /usr/src/app/node_modules/@aws-sdk/middleware-serde/dist/cjs/deserializerMiddleware.js:6:20
    at async /usr/src/app/node_modules/@aws-sdk/middleware-signing/dist/cjs/middleware.js:12:24
    at async StandardRetryStrategy.retry (/usr/src/app/node_modules/@aws-sdk/middleware-retry/dist/cjs/defaultStrategy.js:56:46)
    at async /usr/src/app/node_modules/@aws-sdk/middleware-logger/dist/cjs/loggerMiddleware.js:6:22 {
  '$fault': 'client',
  '$metadata': {
    httpStatusCode: 403,
    requestId: '1f8d9d8e-fe6b-4479-9452-94aaabea22ac',
    extendedRequestId: undefined,
    cfId: undefined,
    attempts: 1,
    totalRetryDelay: 0
  }
}
```

#### Expected behavior
I expect the code to be able to create the EFS AccessPoint

#### Screenshots
NA

**Comment by hardillb** [comment]
If I attach the same role to a plain ubuntu 20.04 container and install the aws cli I see the right role being picked up

**Comment by khant14** [comment]
I've been having the same issues with the V2 SDK. The only workaround I found was to install awscli in the base image. That somehow magically fixes the node sdk to read the Service Account token. If running as a non-root user in the container, you also have to set fsGroup in dockerfile to allow the app user to read the token file.

**Comment by sebsto** [comment]
A couple of stupid question to help to clarify : 
What are the services trusted to assume the role `arn:aws:iam::202058804932:role/eksctl-node-red-addon-iamserviceaccount-defa-Role1-16K0LE4QTM51J` ?
If this is a service role, I assume EKS can service can assume that role, but not necessarily EC2.

To verify : 

```
aws iam get-role --role-name eksctl-node-red-addon-iamserviceaccount-defa-Role1-16K0LE4QTM51J
```

it should return something containing 

```
        "AssumeRolePolicyDocument": {
            "Version": "2012-10-17",
            "Statement": [
                {
                    "Effect": "Allow",
                    "Principal": {
                        "Service": "ec2.amazonaws.com"
                    },
                    "Action": "sts:AssumeRole"
                }
            ]
        },
```

**Comment by hardillb** [comment]
Yep, the env vars are all in scope on the pod

```
$ kubectl exec node-red-multi-tenant-77cb449b4d-qg55n env | grep AWS
kubectl exec [POD] [COMMAND] is DEPRECATED and will be removed in a future version. Use kubectl exec [POD] -- [COMMAND] instead.
Defaulting container name to node-red-manager.
Use 'kubectl describe pod/node-red-multi-tenant-77cb449b4d-qg55n -n default' to see all of the containers in this pod.
AWS_ROLE_ARN=arn:aws:iam::202058804932:role/eksctl-node-red-addon-iamserviceaccount-defa-Role1-16K0LE4QTM51J
AWS_WEB_IDENTITY_TOKEN_FILE=/var/run/secrets/eks.amazonaws.com/serviceaccount/token
AWS_REGION=us-east-2
AWS_DEFAULT_REGION=us-east-2
AWS=true

```

**Comment by sebsto** [comment]
Thanks. And `/var/run/secrets/eks.amazonaws.com/serviceaccount/token` is readable from within the container ?

**Comment by hardillb** [comment]
yep, should be

```
$ kubectl exec --tty --stdin node-red-multi-tenant-77cb449b4d-qg55n --container node-red-manager -- cat /var/run/secrets/eks.amazonaws.com/serviceaccount/token 
eyJhbGciOiJSUzI1NiIsImtpZCI6IjM2ODM5OTU2OTcyOGFiN2EyZTkyMGQ2NzExMjFjMDdhMThhZjliMDkifQ.eyJhdWQiOlsic3RzLmFtYXpvbmF3cy5jb20iXSwiZXhwIjoxNjIwODA1OTI4LCJpYXQiOjE2MjA3MTk1Mjg.....
```
NodeJS app is running as root in the container, no custom user so should be fine

**Comment by sybeck2k** [comment]
it seems that this feat is available since NodeJS [SDK v3.15](https://github.com/aws/aws-sdk-js-v3/releases/tag/v3.15.0) which was just released: can you try update the SDK and run the test again?

**Comment by hardillb** [comment]
v3.15 looks to be working.

Thanks.

Can I suggest an update to the EKS docs (https://docs.aws.amazon.com/eks/latest/userguide/iam-roles-for-service-accounts-minimum-sdk.html) to point to the right version please.
